sql/connect/server/src/main/scala/org/apache/spark/sql/connect/pipelines/PipelinesHandler.scala

```diff
@@ -129,6 +131,46 @@ private[connect] object PipelinesHandler extends Logging {
     }
   }
 
+  /**
+   * Block unsupported SQL commands that are not explicitly allowlisted.
+   */
+  def blockUnsupportedSqlCommand(queryPlan: LogicalPlan): Unit = {
+    val supportedCommand = Set(
+      classOf[DescribeRelation],
+      classOf[ShowTables],
+      classOf[ShowTableProperties],
+      classOf[ShowNamespacesCommand],
+      classOf[ShowColumns],
+      classOf[ShowFunctions],
+      classOf[ShowViews],
+      classOf[ShowCatalogsCommand],
+      classOf[ShowCreateTable])
+    val isSqlCommandExplicitlyAllowlisted = {
+      supportedCommand.exists(c => queryPlan.getClass.getName.equals(c.getName))
+    }
+    val isUnsupportedSqlPlan = if (isSqlCommandExplicitlyAllowlisted) {
+      false
+    } else {
+      // If the SQL command is not explicitly allowlisted, check whether it belongs to
+      // one of commands pipeline explicitly disallow.
+      // If not, the SQL command is supported.
+      queryPlan.isInstanceOf[Command] ||
+      queryPlan.isInstanceOf[CreateTableAsSelect] ||
+      queryPlan.isInstanceOf[CreateTable] ||
+      queryPlan.isInstanceOf[CreateView] ||
+      queryPlan.isInstanceOf[InsertIntoStatement] ||
+      queryPlan.isInstanceOf[RenameTable] ||
+      queryPlan.isInstanceOf[CreateNamespace] ||
+      queryPlan.isInstanceOf[DropView]
+    }
+    // scalastyle:on
```

Comment:
Why scalastyle:on here? Did we turn it off somewhere?

Reply:
Whoops. Good catch!